src/main/java/com/rarchives/ripme/ripper/rippers/FlickrRipper.java

```diff
@@ -250,4 +241,22 @@ public Document getFirstPage() throws IOException {
     public void downloadURL(URL url, int index) {
         addURLToDownload(url, getPrefix(index));
     }
+
+    private URL getLargestImageURL(String imageID, String apiKey) throws IOException {
+        URL imageAPIURL = new URL("https://www.flickr.com/services/rest/?method=flickr.photos.getSizes&api_key=" + apiKey + "&photo_id=" + imageID + "&format=json&nojsoncallback=1");
+        TreeMap<Integer, String> imageURLMap = new TreeMap<>();
+
+        try {
+            JSONArray imageSizes = new JSONObject(Http.url(imageAPIURL).ignoreContentType().get().text()).getJSONObject("sizes").getJSONArray("size");
+            for (int i = 0; i < imageSizes.length(); i++) {
+                JSONObject imageInfo = imageSizes.getJSONObject(i);
+                imageURLMap.put(imageInfo.getInt("width") * imageInfo.getInt("height"), imageInfo.getString("source"));
+            }
+
+        } catch (org.json.JSONException ignore) {
```

Comment:
Don't silently throwaway errors, it makes debugging a pain. Add a logging statement explaining what happened